### PR TITLE
[rllib] Fix torch TD error, IMPALA LR updates

### LINKED ISSUE #9413: [rllib] Raw TD error array shown in torch DQN (ok in TF)
### What is the problem?

```
    last_target_update_ts: 1504
    learner:
      default_policy:
        allreduce_latency: 0.0
        cur_lr: 0.0005
        grad_gnorm: 0.11564214989427604
        max_q: 2.4411678314208984
        mean_q: 1.858547329902649
        mean_td_error: -0.04873969405889511
        min_q: 1.2829080820083618
        td_error: "[ 0.65808916  0.19141054 -0.11309314 -0.46161413 -0.41179168  0.5865724\n\
          \ -0.19835651  0.19526124 -0.30631506  0.03929877 -0.15981436  0.12897038\n\
          \  0.11571825  0.14284062 -0.4166894   0.09555387  0.28290808  0.06396961\n\
          \ -0.13967872 -0.02770591 -0.5273429   0.03402114 -0.3402022   0.00336576\n\
          \  0.09555387 -0.1913867  -0.20986116 -0.31672204  0.55475676 -0.29380023\n\
          \ -0.38737345 -0.24621308]"
```
### Reproduction (REQUIRED)
`rllib train --run=DQN --env=CartPole-v0 --torch -v`

cc @sven1977

### LINKED ISSUE #9428: [rllib] lr_schedule no longer working in IMPALA?
### What is the problem?
I haven't used it in a while, but it seems to me that the `lr_schedule `no longer get's applied when running IMPALA, the attached script works as expected with PPO, but the reported `lr `remains constant for IMPALA, unless it's just a reporting issue.

*Ray version and other system information (Python version, TensorFlow version, OS):*
ray 0.8.6, python 3.6.8, tf 2.2, os rhel 7.6

### Reproduction (REQUIRED)
Please provide a script that can be run to reproduce the issue. The script should have **no external library dependencies** (i.e., use fake or mock data / environments):

If we cannot run your script, we cannot fix your issue.

- [x] I have verified my script runs in a clean environment and reproduces the issue.
- [x] I have verified the issue also occurs with the [latest wheels](https://docs.ray.io/en/latest/installation.html).

[cart-pole-impala.zip](https://github.com/ray-project/ray/files/4908400/cart-pole-impala.zip)

### PR DESCRIPTION
**Pull request by ericl**
## Why are these changes needed?

Closes https://github.com/ray-project/ray/issues/9413

Closes https://github.com/ray-project/ray/issues/9428

### CODE REVIEW

File: rllib/policy/torch_policy.py [hunk 1]
```diff
@@ -39,23 +39,22 @@ class TorchPolicy(Policy):
...
+            model: ModelV2,
+            loss: Callable[[Policy, ModelV2, type, SampleBatch], TensorType],
+            action_distribution_class: TorchDistributionWrapper,
+            action_sampler_fn: Callable[[TensorType, List[TensorType]], Tuple[
+                TensorType, TensorType]] = None,
+            action_distribution_fn: Optional[Callable[[
+                Policy, ModelV2, TensorType, TensorType, TensorType
+            ], Tuple[TensorType, type, List[TensorType]]]] = None,
+            max_seq_len: int = 20,
+            get_batch_divisibility_req: Optional[int] = None):
```
Comment (ericl) [inline rllib/policy/torch_policy.py:57]: Not sure why lint decided to reformat this, I didn't change anything here.

File: rllib/policy/torch_policy.py [hunk 2]
```diff
@@ -228,8 +228,8 @@ def _compute_action_helper(self, input_dict, state_batches, seq_lens,
                         is_training=False)
             else:
                 dist_class = self.dist_class
-                dist_inputs, state_out = self.model(
-                    input_dict, state_batches, seq_lens)
+                dist_inputs, state_out = self.model(input_dict, state_batches,
```
Comment (michaelzhiluo) [inline rllib/policy/torch_policy.py:231]: Looks like Linter is doing the same thing here

File: rllib/agents/impala/tests/test_impala.py [hunk 3]
```diff
@@ -18,6 +18,25 @@ def setUpClass(cls) -> None:
...
+        config = impala.DEFAULT_CONFIG.copy()
+        config["lr_schedule"] = [
+            [0, 0.0005],
+            [10000, 0.000001],
+        ]
+        local_cfg = config.copy()
+        trainer = impala.ImpalaTrainer(config=local_cfg, env="CartPole-v0")
+
+        def get_lr(result):
+            return result["info"]["learner"]["default_policy"]["cur_lr"]
```
Comment (michaelzhiluo) [inline rllib/agents/impala/tests/test_impala.py:31]: This is a bit janky... is there a better to access cur_lr?

File: rllib/agents/impala/impala.py [hunk 4]
```diff
@@ -194,6 +194,8 @@ def __call__(self, item):
             metrics = _get_shared_metrics()
             metrics.counters["num_weight_broadcasts"] += 1
         actor.set_weights.remote(self.weights, _get_global_vars())
+        # Also update global vars of the local worker.
+        self.workers.local_worker().set_global_vars(_get_global_vars())
```
Comment (michaelzhiluo) [inline rllib/agents/impala/impala.py:198]: Does the LR not update without this line of code?

File: rllib/agents/dqn/dqn_torch_policy.py [hunk 5]
```diff
@@ -52,7 +52,6 @@ def __init__(self,
             "mean_q": torch.mean(q_t_selected),
             "min_q": torch.min(q_t_selected),
             "max_q": torch.max(q_t_selected),
-            "td_error": self.td_error,
```
Comment (sven1977) [inline rllib/agents/dqn/dqn_torch_policy.py:55]: yeah, this was annoying. Thanks for removing it.

File: rllib/agents/impala/impala.py (hunk 4 above)
Comment (sven1977) [reply rllib/agents/impala/impala.py:198]: I think this was the root-cause of this issue: That the local worker's on_global_var_update was never called. The remote workers' lr was always fine and updated correctly.

File: rllib/policy/torch_policy.py (hunk 1 above)
Comment (sven1977) [reply rllib/policy/torch_policy.py:57]: np, actually: This looks nicer :)

**Review by sven1977 — CHANGES_REQUESTED** [review]
All good, just the tests are still failing with e.g.:
```
AttributeError: 'PPOTorchPolicy' object has no attribute '_optimizers'
```

File: rllib/agents/impala/impala.py (hunk 4 above)
Comment (sven1977) [reply rllib/agents/impala/impala.py:198]: So, yes, w/o this line, lr would not get updated on the driver.

File: rllib/agents/impala/tests/test_impala.py (hunk 3 above)
Comment (sven1977) [reply rllib/agents/impala/tests/test_impala.py:31]: Fine with me, but yeah: `trainer.get_policy().cur_lr` would be shorter.

**Review by sven1977 — APPROVED** [review]
Looks good. Just please make sure the tests are ok.

File: rllib/agents/impala/impala.py (hunk 4 above)
Comment (michaelzhiluo) [reply rllib/agents/impala/impala.py:198]: Got it! This is the core line of the LR PR :).

**Pull request merged**